### PR TITLE
feat: export compiler as dual CJS / ESM package

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

Use `tsup` to distribute the compiler as both CJS and ESM, so people (me) downstream can use it fully

## Testing

Tests should still pass + tested manually in prettier plugin in this PR: https://github.com/withastro/prettier-plugin-astro/pull/349

## Docs

N/A

### CODE REVIEW

**Comment by Princesseuh** [comment]
!preview dual

File: packages/compiler/package.json [hunk 1]
```diff
@@ -7,38 +7,51 @@
...
+      "types": "./dist/node/utils.d.ts",
+      "browser": "./dist/browser/utils.js",
+      "import": "./dist/node/utils.js",
+      "require": "./dist/node/utils.cjs",
+      "default": "./dist/browser/utils.js"
     },
-    "./astro.wasm": "./astro.wasm",
-    "./types": "./types.d.ts"
+    "./astro.wasm": "./dist/astro.wasm",
+    "./types": "./dist/shared/types.d.ts"
```
Comment (natemoo-re) [inline packages/compiler/package.json:46]: Not strictly required but I like to make sure this is always exposed. Surprised we didn't already have that!

```suggestion
    "./package.json": "./package.json"
```

File: .changeset/kind-icons-attack.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/compiler': minor
```
Comment (natemoo-re) [inline .changeset/kind-icons-attack.md:2]: I think `minor` is "nicer," but... this is technically a `major` breaking change, right? Version numbers are free!

**Review by natemoo-re — COMMENTED** [review]
Looks great! Just had a few comments.

File: .changeset/kind-icons-attack.md (hunk 2 above)
Comment (natemoo-re) [reply .changeset/kind-icons-attack.md:2]: Discussed on Discord and we decided to go with `minor`.

**Review by natemoo-re — APPROVED** [review]
Approved. Super clean, thanks for tackling this!

**Pull request merged**